src/core/public/chrome/ui/header/collapsible_nav_group_enabled.tsx

```diff
@@ -0,0 +1,347 @@
+      link,
+      appId,
+      dataTestSubj: `collapsibleNavAppLink-${link.id}`,
+      navigateToApp,
+      onClick: (event) => {
+        onNavItemClick(event, link);
+      },
+    });
+
+    return {
+      id: `${link.id}-${link.title}`,
+      name: <EuiText size="s">{link.title}</EuiText>,
+      onClick: euiListItem.onClick,
+      href: euiListItem.href,
+      emphasize: euiListItem.isActive,
+      className: `nav-link-item ${className || ''}`,
+      buttonClassName: 'nav-link-item-btn',
+      'data-test-subj': euiListItem['data-test-subj'],
+      'aria-label': link.title,
+    };
+  };
+  const createSideNavItem = (navLink: LinkItem, className?: string): EuiSideNavItemType<{}> => {
+    if (navLink.itemType === LinkItemType.LINK) {
+      if (navLink.link.title === titleForSeeAll) {
+        const navItem = createNavItem({
+          link: navLink.link,
+        });
+
+        return {
+          ...navItem,
+          name: <EuiText color="success">{navItem.name}</EuiText>,
+          emphasize: false,
+        };
+      }
+
+      return createNavItem({
+        link: navLink.link,
+        className,
+      });
+    }
+
+    if (navLink.itemType === LinkItemType.PARENT_LINK && navLink.link) {
+      return {
+        ...createNavItem({ link: navLink.link }),
+        forceOpen: true,
+        items: navLink.links.map((subNavLink) => createSideNavItem(subNavLink, 'nav-nested-item')),
+      };
+    }
+
+    if (navLink.itemType === LinkItemType.CATEGORY) {
+      return {
+        id: navLink.category?.id ?? '',
+        name: <div className="nav-link-item">{navLink.category?.label ?? ''}</div>,
+        items: navLink.links?.map((link) => createSideNavItem(link)),
+        'aria-label': navLink.category?.label,
+      };
+    }
+
+    return {} as EuiSideNavItemType<{}>;
+  };
+  const orderedLinksOrCategories = getOrderedLinksOrCategories(navLinks);
+  const sideNavItems = orderedLinksOrCategories
+    .map((navLink) => createSideNavItem(navLink))
+    .filter((item): item is EuiSideNavItemType<{}> => !!item);
+  return (
+    <EuiFlexItem style={style}>
+      <EuiSideNav items={sideNavItems} />
+      {suffix}
+    </EuiFlexItem>
+  );
+}
+
+export function CollapsibleNavGroupEnabled({
+  basePath,
+  id,
+  isLocked,
+  isNavOpen,
+  storage = window.localStorage,
+  onIsLockedUpdate,
+  closeNav,
+  navigateToApp,
+  navigateToUrl,
+  logos,
+  setCurrentNavGroup,
+  ...observables
+}: CollapsibleNavGroupEnabledProps) {
+  const navLinks = useObservable(observables.navLinks$, []).filter((link) => !link.hidden);
+  const appId = useObservable(observables.appId$, '');
+  const navGroupsMap = useObservable(observables.navGroupsMap$, {});
+  const currentNavGroup = useObservable(observables.currentNavGroup$, undefined);
+
+  const onGroupClick = (
+    e: React.MouseEvent<HTMLButtonElement, MouseEvent>,
+    group: NavGroupItemInMap
+  ) => {
+    const fulfilledLinks = fulfillRegistrationLinksToChromeNavLinks(
```

Comment:
what does fullfilledLinks mean?

Reply:
Actually the navLinks inside navGroupMap has only partial of the required field to construct a `ChromeNavLink`, so we call the fulfill function here to get the detailed information by combining the info from ChromeRegistrationNavLink and ChromeNavLink. You can regard it as a `left join`.